src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -53,8 +53,14 @@ private IReadOnlyList<RestoreSummaryRequest> GetRequestsFromItems(RestoreArgs re
                 throw new ArgumentNullException(nameof(dgFile));
             }
 
-            // Validate the dg file input, this throws if errors are found.
-            SpecValidationUtility.ValidateDependencySpec(dgFile);
+            // When the project has an parsing error, for example an invalid TFM or version, we expect a fake dgSpec,
+            // in order to getnerate the assets file. Downstream tools, particularly Visual Studio's project system
+            // expects an assets file with errors, not exceptions being thrown.
+            if (restoreContext.AdditionalMessages == null || !restoreContext.AdditionalMessages.Any(m => m.Level == Common.LogLevel.Error))
+            {
+                // Validate the dg file input, this throws if errors are found.
+                SpecValidationUtility.ValidateDependencySpec(dgFile);
```

Comment:
I don't think the spec validation should be failing?

Reply:
[It validates project TFMs](https://github.com/NuGet/NuGet.Client/blob/3803820961f4d61c06d07b179dab1d0439ec0d91/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/SpecValidationUtility.cs#L131-L162). When the TFM in the project file is invalid, I didn't want to hardcode a TFM just to make the validation pass, as that'll get written to the assets file and I didn't want the assets file to list a TFM that wasn't specified in the project file.

Reply:
:( 
Can we leave the validation for the other projects? Is that a consideration?
Maybe in a future PR, don't think it's worth blocking this PR on that.